cmd/snap-confine/snap-confine.c

```diff
@@ -501,9 +501,9 @@ int main(int argc, char **argv)
 			die("setuid failed");
 
 		if (real_gid != 0 && (getuid() == 0 || geteuid() == 0))
-			die("permanently dropping privs did not work");
+			die("permanently dropping privs did not work. you must use sudo but you must NOT be logged in as root.");
```

Comment:
Can you describe how you triggered this error?

Reply:
@bboozzoo added reproduction steps to the PR description

Reply:
@SeanColombo, I tried your reproducer on a 20.04 system and couldn't reproduce, could you maybe open a forum topic with details of like `snap version` and other details about this system so we can look into why this happened for you? 

we don't typically like to use github to debug things like this as it's less visible to the wider community who may have other answers for you, typically only the snapd devs will look at these pr's